src/components/organisms/Navbar/Navbar/Navbar.tsx

```diff
@@ -1,84 +1,199 @@
+    margin-right: ${spacing[10]};
+  }
 `;
 
-export interface INavbarProps {
-  /**
-   * background color
-   */
-  backgroundColor?: string;
-  /**
-   * Component appearing on the left
-   */
-  left?: React.ReactNode;
-  /**
-   * Component appearing in the center
-   */
-  center?: React.ReactNode;
-  /**
-   * Component appearing on the right
-   */
-  right?: React.ReactNode;
-}
+const IconContainer = styled.span`
+  display: flex;
+  height: ${mobileNavbarHeight}px;
+  width: ${mobileNavbarHeight}px;
+  font-size: 24px;
+  justify-content: center;
+  align-items: center;
+`;
+
+const HamburgerContainer = styled(IconContainer)<IHamburgerContainerProps>`
+  background-color: ${({ open }) => (open ? colors.white : 'transparent')};
+`;
+
+const HamburgerMenu = styled.aside<IHamburgerContainerProps>`
+  position: absolute;
+  right: 0;
+  top: ${mobileNavbarHeight}px;
+  bottom: 0;
+  left: 0;
+
+  display: flex;
+  flex-direction: column;
+  min-height: 100vh;
+  width: 100%;
+  padding: ${spacing[8]} ${spacing[4]} 0;
+
+  background: ${colors.white};
+
+  transition: transform 0.25s ease-in-out;
+  transform: ${({ open }) => (open ? 'translateX(0)' : 'translateX(-100%)')};
+  z-index: 2;
+  overflow-y: auto;
+`;
+
+const LargeDeviceNavbar: React.FC<ILargeDeviceNavbar> = ({ children, overlayLogoWith }) => {
+  return (
+    <>
+      <LogoContainer>
+        <Logo color={colors.brand} width="150px" negative />
+        {overlayLogoWith}
+      </LogoContainer>
+      <NavbarLinksListContainer>{children}</NavbarLinksListContainer>
+    </>
+  );
+};
 
-const Navbar = ({ backgroundColor = colors.white, left, center, right }: INavbarProps) => {
-  const overThreshold = useScrollThreshold();
+const SmallDeviceNavbar: React.FC<INavbarProps> = ({ children, overlayLogoWith, withCTA, cta }) => {
+  const [open, setOpen] = useState<boolean>(false);
 
   return (
     <>
-      <Spacer />
-      <LayoutOuter backgroundColor={backgroundColor} overlap={overThreshold}>
-        <Container>
-          <LayoutInner>
-            {left && <div>{left}</div>}
-            {center && <div>{center}</div>}
-            {right && <div>{right}</div>}
-          </LayoutInner>
-        </Container>
-      </LayoutOuter>
+      {children ? (
+        <HamburgerContainer open={open} onClick={() => setOpen(!open)}>
+          <Icon variant={faBars} color={open ? colors.brand : colors.white} fixedWidth />
+        </HamburgerContainer>
+      ) : (
+        <IconContainer />
+      )}
+      <LogoContainer>
+        <Logo color={colors.brand} width="150px" negative />
+        {overlayLogoWith}
+      </LogoContainer>
+      <IconContainer>{withCTA && cta}</IconContainer>
+      {children && <HamburgerMenu open={open}>{children}</HamburgerMenu>}
     </>
   );
 };
 
-export default Navbar;
+const NavbarWrapper: React.FC<INavbarProps> = ({
+  children,
+  overlayLogoWith = <a href={process.env.REACT_APP_ZOPA_LOGO_LINK || 'https://www.zopa.com'} />,
```

Comment:
I don't think we add env variables in our react components. Do we need it? I would keep the default with 'https://www.zopa.com' only. IF we need to modify this maybe we can use another prop? Same for Navbar action

Reply:
I didn't want to add an additional prop and know we have spoken about having a common list of env variables that we all extend from. Prob got ahead of myself and added it in without discussing with others